### PR TITLE
chore(TS): Text SVG export mixin

### PR DESCRIPTION
**Pull request by ShaMan123**
## Description

Last file accept canvas files

## Changes

- `src/mixins/itext.svg_export.ts`:
The diff is very harsh... Perhaps better to view commit by commit? Since the first commit is the transfomer's work and the second is just generic migration and moving a method or 2.
Renamed to `src/mixins/text.svg_export.ts` since Github doesn't pick it up because the diff is too large. Revert at will, ebbbffd5e
- `src/mixins/object.svg_export.ts`: cleanup
- `src/shapes/itext.class.ts`: jsdoc
- `src/util/misc/svgParsing.ts`: extract utils to here

### CODE REVIEW

File: src/mixins/object.svg_export.ts [hunk 1]
```diff
@@ -170,25 +177,6 @@ export class FabricObjectSVGExportMixin {
     return `${svgTransform}${additionalTransform}" `;
   }
 
-  _setSVGBg(textBgRects: string[]) {
```
Comment (ShaMan123) [inline src/mixins/object.svg_export.ts:173]: used by text, refactored into text svg export mixin

File: src/mixins/object.svg_export.ts [hunk 2]
```diff
@@ -5,11 +5,18 @@ import { uid } from '../util/internals/uid';
...
 
-function getSvgColorString(prop: string, value?: any) {
+/**
+ * Adobe Illustrator (at least CS5) is unable to render rgba()-based fill values
+ * we work around it by "moving" alpha channel into opacity attribute and setting fill's alpha to 1
+ * @param prop
+ * @param value
+ * @returns
+ */
+export function getSvgColorString(prop: string, value?: any) {
```
Comment (ShaMan123) [inline src/mixins/object.svg_export.ts]: now used by Text instead of `_getFillAttributes`

File: index.js [hunk 3]
```diff
@@ -55,8 +55,8 @@ import './src/filters/gamma_filter.class'; // optional image_filters
 import './src/filters/composed_filter.class'; // optional image_filters
 import './src/filters/hue_rotation.class'; // optional image_filters
 import './src/shapes/text.class'; // optional text
+import './src/mixins/itext.svg_export'; // optional svg
```
Comment (ShaMan123) [inline index.js]: Naming is wrong, it is a Text mixin. So it should be applied before subclassing IText

File: src/mixins/itext.svg_export.ts [hunk 4]
```diff
@@ -1,336 +1,303 @@
...
-       * Returns SVG representation of an instance
-       * @param {Function} [reviver] Method for further parsing of svg representation.
-       * @return {String} svg representation of an instance
-       */
-      _toSVG: function () {
-        var offsets = this._getSVGLeftTopOffsets(),
-          textAndBg = this._getSVGTextAndBg(offsets.textTop, offsets.textLeft);
-        return this._wrapSVGTextAndBg(textAndBg);
-      },
+function createSVGRect(
```
Comment (ShaMan123) [inline src/mixins/itext.svg_export.ts]: formally `_setSVGBg` and `_pushTextBgRect`

File: src/mixins/object.svg_export.ts [hunk 5]
```diff
@@ -2,32 +2,13 @@
...
-import { matrixToSVG } from '../util/misc/svgParsing';
+import { colorPropToSVG, matrixToSVG } from '../util/misc/svgParsing';
 import { toFixed } from '../util/misc/toFixed';
 
-type SVGReviver = (markup: string) => string;
+export type SVGReviver = (markup: string) => string;
 
 /* _TO_SVG_START_ */
 
-function getSvgColorString(prop: string, value?: any) {
```
Comment (ShaMan123) [inline src/mixins/object.svg_export.ts:12]: extracted to util and renamed to `colorPropToSVG`

**Review by ShaMan123 — COMMENTED** [review]
READY

renamed `src/mixins/itext.svg_export.ts` to `src/mixins/text.svg_export.ts`
Github doesn't pick it up because the diff is too large. Revert at will, ebbbffd5e

File: src/mixins/text.svg_export.ts [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+          boxWidth,
+          heightOfLine
+        )
+      );
+  }
+
+  /**
+   * @deprecated unused
+   */
+  _getSVGLineTopOffset(lineIndex: number) {
```
Comment (ShaMan123) [inline src/mixins/text.svg_export.ts:268]: this method is dead
can we remove it?

**Review by ShaMan123 — COMMENTED** [review]
READY

**Review by asturur — APPROVED** [review]

**Pull request merged**